dex/src/main/scala/com/wavesplatform/dex/AddressWsMutableState.scala

```diff
@@ -63,9 +63,26 @@ case class AddressWsMutableState(activeWsConnections: Map[ActorRef, UUID],
     )
   }
 
+  def sendSnapshot(balances: Map[Asset, WsBalances], orders: Seq[WsOrder]): Unit = {
+    pendingWsConnections.keys.foreach(_ ! WsAddressState(balances, orders, 0))
+  }
+
+  def sendDiffs(balances: Map[Asset, WsBalances], orders: Seq[WsOrder]): AddressWsMutableState = copy(
+    activeWsConnections = activeWsConnections.map { // dirty but linear
```

Comment:
About comment: it wouldn't be a quadratic or worse even we run such loop N times. But we do N-1 more unnecessary work :)